tasks/post_to_catalog/post_to_catalog.py

```diff
@@ -0,0 +1,241 @@
+        LOGGER.error(
+            "Error while posting provider '{provider_id}', collection '{collection_id}', "
+            "granule '{cumulus_granule_id}' to inventory: {sql_ex}",
+            provider_id=provider["providerId"],
+            collection_id=collection["collectionId"],
+            cumulus_granule_id=granule["cumulusGranuleId"],
+            sql_ex=sql_ex,
+        )
+        raise
+
+
+def create_provider_sql():
+    return text(
+        """
+        INSERT INTO providers
+            ("provider_id", "name")
+        VALUES
+            (:provider_id, :name)
+        ON CONFLICT DO NOTHING
+        """
+    )
+
+
+def create_collection_sql():
+    return text(
+        """
+    INSERT INTO collections
+            ("collection_id", "shortname", "version")
+    VALUES
+        (:collection_id, :shortname, :version)
+    ON CONFLICT DO NOTHING
+    """
+    )
+
+
+def create_granule_sql():
+    return text(
+        """
+    INSERT INTO granules
+        ("collection_id", "cumulus_granule_id", "execution_id", "ingest_time", "cumulus_create_time", "last_update")
+    VALUES
+        (:collection_id, :cumulus_granule_id, :execution_id, :ingest_time, :cumulus_create_time, :last_update)
+    ON CONFLICT ("collection_id", "cumulus_granule_id") DO UPDATE
+        SET "execution_id"=:execution_id, "last_update"=:last_update
+    RETURNING id"""
+    )
+    # ON CONFLICT will only trigger if both collection_id and cumulus_granule_id match.
+
+
+def create_file_sql():
+    return text(
+        """
+    INSERT INTO files
+        ("granule_id", "name", "orca_archive_location", "cumulus_archive_location", "key_path", "ingest_time", "etag", 
+        "version", "size_in_bytes", "hash", "hash_type")
+    VALUES
+        (:granule_id, :name, :orca_archive_location, :cumulus_archive_location, :key_path, :ingest_time, :etag, 
+        :version, :size_in_bytes, :hash, :hash_type)
+    ON CONFLICT ("cumulus_archive_location", "key_path") DO UPDATE
+        SET "granule_id"=:granule_id, "name"=:name, "orca_archive_location"=:orca_archive_location, 
+        "ingest_time"=:ingest_time, "etag"=:etag, "version"=:version, "size_in_bytes"=:size_in_bytes, "hash"=:hash, 
+        "hash_type"=:hash_type"""
+    )
+    # ON CONFLICT will only trigger if all listed properties match.
+
+
+def handler(event: Dict[str, List], context) -> None:
+    """
+    Lambda handler. Receives a list of queue entries from an SQS queue, and posts them to a database.
+
+    Args:
+        event: A dict with the following keys:
+            'Records' (List): A list of dicts with the following keys:
+                'messageId' (str)
+                'receiptHandle' (str)
+                'body' (str): A json string representing a dict.
+                    See catalog_record_input in schemas for details.
+        context: An object passed through by AWS. Used for tracking.
+    Environment Vars: See shared_db.py's get_configuration for further details.
```

Comment:
Probably a copy/paste issue here but the information here doesn't seem correct. For environment variables you likely only need PREFIX and AWS_REGION which we check for in the shared library.